### PR TITLE
TOC, tools/dm: add the DM upgrade doc

### PR DESCRIPTION
**Pull request by CaitinChen**
@csuzhangxc @lilin90 PTAL

### CODE REVIEW

File: tools/dm/dm-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+---
+title: DM 版本升级
+category: tools
+---
+
+# DM 版本升级
+
+本文档主要介绍各不完全兼容的 DM (Data Migration) 版本间的升级操作步骤。
+
+假设依时间先后顺序存在 V-A、V-B、V-C 等 3 个互不兼容的版本，现在需要从 V-A 升级到 V-C。定义从 V-A 升级到 V-B 的操作为 Upgrade-A-B，从 V-B 升级到 V-C 的操作为 Upgrade-B-C。
```
Comment (csuzhangxc) [inline tools/dm/dm-upgrade.md]:
```suggestion
假设依时间先后顺序存在 V-A、V-B、V-C 3 个互不兼容的版本，现在需要从 V-A 升级到 V-C。定义从 V-A 升级到 V-B 的操作为 Upgrade-A-B，从 V-B 升级到 V-C 的操作为 Upgrade-B-C。
```

File: tools/dm/dm-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+### 升级操作示例
+
+1. 变更 `inventory.ini` 的配置信息。
+
+    - 为所有 DM-worker 实例设置对应的 `source_id`。
+    
+    如将
+
+    ```bash
```
Comment (amyangfei) [inline tools/dm/dm-upgrade.md]:
```suggestion
    ```ini
```

File: tools/dm/dm-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    dm-worker1 ansible_host=172.16.10.72 server_id=101 mysql_host=172.16.10.72 mysql_user=root mysql_password='VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU=' mysql_port=3306
+    ```
+
+    变更为
+
+    ```bash
```
Comment (amyangfei) [inline tools/dm/dm-upgrade.md]:
```suggestion
    ```ini
```

File: tools/dm/dm-upgrade.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+### 升级操作示例
+
+1. 变更 `inventory.ini` 配置信息。
+
+    - 移除所有 `dm_worker_status_port` 配置项，根据需要变更 `dm_worker_port` 配置项。
+    - 移除所有 `dm_master_status_port` 配置项，根据需要变更 `dm_master_port` 配置项。
+    
+    如将
+    
+    ```bash
```
Comment (amyangfei) [inline tools/dm/dm-upgrade.md]:
```suggestion
    ```ini
```

File: tools/dm/dm-upgrade.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    dm_worker1_1 ansible_host=172.16.10.72 server_id=101 deploy_dir=/data1/dm_worker dm_worker_port=10081 dm_worker_status_port=10082 mysql_host=172.16.10.81 mysql_user=root mysql_password='VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU=' mysql_port=3306
+    ```
+    
+    变更为
+    
+    ```bash
```
Comment (amyangfei) [inline tools/dm/dm-upgrade.md]:
```suggestion
    ```ini
```

File: tools/dm/dm-upgrade.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    dm_worker1_1 ansible_host=172.16.10.72 server_id=101 deploy_dir=/data1/dm_worker dm_worker_port=8262 mysql_host=172.16.10.81 mysql_user=root mysql_password='VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU=' mysql_port=3306
+    ```
+    
+    将
+    
+    ```bash
```
Comment (amyangfei) [inline tools/dm/dm-upgrade.md]:
```suggestion
    ```ini
```

File: tools/dm/dm-upgrade.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+    dm_master ansible_host=172.16.10.71 dm_master_port=12080 dm_master_status_port=12081	
+    ```
+    
+    变更为
+    
+    ```bash
```
Comment (amyangfei) [inline tools/dm/dm-upgrade.md]:
```suggestion
    ```ini
```

File: tools/dm/dm-upgrade.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+- `config`：上游 MySQL 的地址、用户名、密码等。
+- `instance-id`：标识一个上游 MySQL。
+
+从此版本开始，上述两项配置信息被移除，并增加了如下配置信息：
+
+- `source_id`：存在于 `inventory.ini` 中，用于标识一个上游 MySQL 实例或一个主从复制组。
+- `source-id`：存在于任务配置文件的 `mysql-instances` 中，其取值与 `inventory.ini` 中的 `source_id` 对应。
+
+> **注意：** 如果需要确保已有任务存储在下游数据库的断点信息能继续被使用，`source_id`/`source-id` 的值需要与对应 DM-worker 变更前的 `instance-id` 一致。
```
Comment (lilin90) [inline tools/dm/dm-upgrade.md]:
```suggestion
> **注意**：如果需要确保已有任务存储在下游数据库的断点信息能继续被使用，`source_id`/`source-id` 的值需要与对应 DM-worker 变更前的 `instance-id` 一致。
```

**Comment by CaitinChen** [comment]
@csuzhangxc @amyangfei @lilin90 Thanks.   @lilin90 PTAL again.

**Comment by csuzhangxc** [comment]
LGTM

**Comment by amyangfei** [comment]
LGTM

File: tools/dm/dm-upgrade.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+- `source_id`：存在于 `inventory.ini` 中，用于标识一个上游 MySQL 实例或一个主从复制组。
+- `source-id`：存在于任务配置文件的 `mysql-instances` 中，其取值与 `inventory.ini` 中的 `source_id` 对应。
+
+> **注意**：如果需要确保已有任务存储在下游数据库的断点信息能继续被使用，`source_id`/`source-id` 的值需要与对应 DM-worker 变更前的 `instance-id` 一致。
+
+### 升级操作示例
+
+1. 变更 `inventory.ini` 的配置信息。
+
+    - 为所有 DM-worker 实例设置对应的 `source_id`。
```
Comment (lilin90) [inline tools/dm/dm-upgrade.md]:
```suggestion
    为所有 DM-worker 实例设置对应的 `source_id`。
```

File: tools/dm/dm-upgrade.md [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+
+    ```ini
+    dm-worker1 ansible_host=172.16.10.72 source_id="mysql-replica-01" server_id=101 mysql_host=172.16.10.72 mysql_user=root mysql_password='VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU=' mysql_port=3306
+    ```
+
+2. 使用 DM-Ansible 滚动升级 DM。
+
+3. 变更任务配置文件 (`task.yaml`)。
+    
+    - 移除其中的 `config` 与 `instance-id` 配置项，增加 `source-id` 配置项（与 `inventory.ini` 中的 `source_id` 对应）。
```
Comment (lilin90) [inline tools/dm/dm-upgrade.md]:
```suggestion
    移除其中的 `config` 与 `instance-id` 配置项，增加 `source-id` 配置项（与 `inventory.ini` 中的 `source_id` 对应）。
```

**Comment by CaitinChen** [comment]
@lilin90 Thanks. PTAL again.

**Pull request merged**